### ISSUE TITLE
nettle header files missing on OpenWrt, even though they're not

### ISSUE THREAD

**Issue by karlic** [pauldreik/rdfind#160]
```
> wget https://ftp.gnu.org/gnu/nettle/nettle-3.4.1.tar.gz
> ls nettle-3.4.1.tar.gz
> nettle-3.4.1.tar.gz
> gunzip nettle-3.4.1.tar.gz
> tar -xf nettle-3.4.1.tar
> ls
> nettle-3.4.1 rdfind
> pwd
> /root
> cd rdfind
> ln -s /root/nettle-3.4.1 nettle
> ./configure
...
> checking for nettle/sha.h... no
> configure: error: 
 nettle header files missing. Please install nettle
 first. If you have already done so and get this error message
 anyway, it may be installed somewhere else, maybe because you
 don't have root access. Pass CPPFLAGS=-I/your/path/to/nettle to configure
 and try again. The path should be so that \#include "nettle/sha.h" works.
 On Debian-ish systems, use "apt-get install nettle-dev" to get a system
 wide nettle install.

```
Passing CPPFLAGS doesn't seem to work either
BTW, first line of sha.h says it's deprecated.

**Comment by pauldreik** [comment]
did you configure and build nettle? there is no sign of that in the text above.

**Comment by karlic** [comment]
Yes. I had already installed nettle using openwrt's opkg and thought I just needed the headers. When that didn't work, I tried building and installing nettle, but that made no difference.

**Comment by pauldreik** [comment]
is there something like a nettle-dev package which contains the headers?

**Comment by karlic** [comment]
I don't think so. So I used wget to get them from https://ftp.gnu.org/gnu/nettle/nettle-3.4.1.tar.gz

**Comment by pauldreik** [comment]
this sounds like you are building inside openwrt - is that right?

I read at https://openwrt.org/docs/guide-developer/toolchain/building_openwrt_on_openwrt#setting_up_the_development_environment which might be useful in that case.

It would be cool if rdfind would be available on openwrt as a package!

**Comment by karlic** [comment]
Yes, I'm trying to build in openwrt. I have an extroot (storage off the device) so have the space. It would indeed be cool to have a rdfind package. My next step will be to set up a build environment if I can't get on-device building working.

**Comment by raenye** [comment]
> It would be cool if rdfind would be available on openwrt as a package!

https://github.com/openwrt/packages/pull/26905
